740. Delete and Earn

```diff
@@ -0,0 +1,39 @@
+Time complexity: O(n)+O(max of array)
+Space complexity: O(n)+O(max of array)
```

Comment:
Can you explain the space complexity? It should be O(max of array) only. Why adding O(n)?